### DISCUSSION TITLE
deepmd: nonhamiltonian (far from equilibrium) systems, training on forces and coordinates only

### DISCUSSION THREAD

**Discussion by chtchelkatchev** [deepmodeling/deepmd-kit#2882] [Ideas]
Hello! 
I tried to use deepmd to describe plasma. Unfortunately, this did not work out. In a plasma under the influence of an alternating external field, taking into account dissipative effects, there is no Hamiltonian and potential energy (what is called energy in deepmd). Deepmd training in this case is possible only by forces and coordinates. Attempts to use deepmd in this mode were unsuccessful. Only when we turned off dissipation and external fields (the system turned into a Hamiltonian one), deepmd training on forces and coordinates was successful and deepmd predicted correct results in lammps. We kindly ask you to consider the possibility of training by coordinates and forces in future versions of deepmd. Perhaps this will be useful not only for plasma: as I understand it, forces are calculated by deepmd using tf.gradients, and differentiation is an inexact operation in numerical calculations, which is better to get rid of...

I also had to deal with situations in another area when, due to an insufficient mesh of k-points, the energies calculated in the VASP were not accurate, but the forces were already quite accurate. The result was a systematic discrepancy between the results obtained in LAMPS+DeepMD and VASP. It would be nice to make sure that at the output the deepmd immediately produces forces, without calculating or using energy at the input. 
Thank you in advance for your response.

**Comment by njzjz** [comment]
The method `tf.gradients` calculates analytical rather than numerical gradients. If you mean fitting a property that is not the negative gradient of the potential energy, see #2617 in this case.

If the property is affected by the external field, it may need to be added to the descriptor, as it breaks equivalent. @wanghan-iapcm may give some suggestions in this case.

**Reply by chtchelkatchev** [reply]
Thank You. I'm trying this.
